oe_whitelabel.theme

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * @file
+ * Functions to support theming.
+ */
+
+declare(strict_types = 1);
+
+/**
+ * Implements hook__preprocess_links__language_block().
+ */
+function oe_whitelabel_preprocess_links__language_block(&$variables) {
+  /** @var \Drupal\Core\Language\LanguageInterface[] $original_languages */
+  $original_languages = \Drupal::service('language_manager')->getNativeLanguages();
+  $currentLanguage = \Drupal::languageManager()->getCurrentLanguage()->getName();
+  $variables['language']['link'] = [
+    'label' => $currentLanguage,
+    'href' => $currentLanguage,
+    'target' => 'languageModal',
+  ];
+
+  $eu_links = [];
+  $non_eu_links = [];
+  foreach ($original_languages as $language) {
+    $url = $variables['links'][$language->getId()]['link']['#url'];
+    $host = \Drupal::request()->getHost();
+
```

Comment:
construct a link object instead of initializing almost the same render array twice
```suggestion
$link = [
      'href' => $href,
      'name' => $label,
      'id' => 'link_' . $language_code,
      'hreflang' => $language_code,
    ];
```